src/us/kbase/workspace/test/database/provenance/CreditMetadataTest.java

```diff
@@ -0,0 +1,655 @@
+			final List<Title> titles) {
+		assertCreditMetadataFields(creditMetadata, identifier, rt, expectedMap, ELC, ELD, ELF, ELRI, titles);
+	}
+
+	private void assertCreditMetadataFieldsWithDates(
+			final CreditMetadata creditMetadata,
+			final String identifier,
+			final ResourceType rt,
+			final List<Title> titles) {
+		assertCreditMetadataFields(creditMetadata, identifier, rt, EM, ELC, DATE_LIST, ELF, ELRI, titles);
+	}
+
+	@Test
+	public void buildMinimal() throws Exception {
+		for (final ResourceType rt : ResourceType.values()) {
+			for (Map.Entry<String, String> entry : VALID_PID_MAP.entrySet()) {
+				final CreditMetadata creditMetadata = CreditMetadata
+						.getBuilder(entry.getKey(), rt, TITLE_LIST)
+						.withVersion(VERSION_STRING)
+						.build();
+				assertCreditMetadataFields(creditMetadata, entry.getValue(), rt, VERSION_MAP, ELC, ELD, ELF, ELRI, TITLE_LIST);
+			}
+		}
+	}
+
+	@Test
+	public void buildMinimalResourceTypeStrings() throws Exception {
+		for (final String rts : VALID_RESOURCE_TYPE_STRINGS) {
+			for (Map.Entry<String, String> entry : VALID_PID_MAP.entrySet()) {
+				final CreditMetadata creditMetadata = CreditMetadata
+						.getBuilder(entry.getKey(), rts, TITLE_LIST)
+						.withVersion(VERSION_STRING)
+						.build();
+				assertCreditMetadataFields(creditMetadata, entry.getValue(), ResourceType.DATASET, VERSION_MAP, ELC, ELD, ELF, ELRI, TITLE_LIST);
+			}
+		}
+	}
+
+	@Test
+	public void buildMinimalDefaultResourceType() throws Exception {
+		final CreditMetadata creditMetadata = CreditMetadata
+				.getBuilder(IDENTIFIER_STRING, (String) null, TITLE_LIST)
+				.withVersion(VERSION_STRING)
+				.build();
+
+		assertCreditMetadataFields(creditMetadata, IDENTIFIER_STRING, ResourceType.DATASET, VERSION_MAP,
+				ELC, ELD, ELF, ELRI, TITLE_LIST);
+
+
+		final CreditMetadata creditMetadata2 = CreditMetadata
+				.getBuilder(IDENTIFIER_STRING, (ResourceType) null, TITLE_LIST)
+				.withVersion(VERSION_STRING)
+				.build();
+
+		assertCreditMetadataFields(creditMetadata2, IDENTIFIER_STRING, ResourceType.DATASET, VERSION_MAP,
+				ELC, ELD, ELF, ELRI, TITLE_LIST);
+
+	}
+
+	@Test
+	public void buildCleanAndTrimComments() throws Exception {
+		for (final ResourceType rt : ResourceType.values()) {
+			final CreditMetadata creditMetadata = CreditMetadata
+					.getBuilder(IDENTIFIER_STRING, rt, TITLE_LIST)
+					.withComments(COMMENTS_STRING_UNTRIMMED)
+					.withVersion(VERSION_STRING_UNTRIMMED)
+					.build();
+
+			assertCreditMetadataFields(creditMetadata, IDENTIFIER_STRING, rt, COMMENTS_VERSION_MAP,
+					ELC, ELD, ELF, ELRI, TITLE_LIST);
+		}
+	}
+
+	@Test
+	public void buildCleanURLTypeLicense() throws Exception {
+		for (final ResourceType rt : ResourceType.values()) {
+			final CreditMetadata creditMetadata = CreditMetadata
+					.getBuilder(IDENTIFIER_STRING, rt, TITLE_LIST)
+					.withLicense(LICENSE_URL_STRING)
+					.withVersion(VERSION_STRING_UNTRIMMED)
+					.build();
```

Comment:
TODO: add a couple more examples here using `VALID_URL_LIST` in `[src/us/kbase/workspace/test/database/provenance/ProvenanceTestCommon.java](https://github.com/kbase/workspace_deluxe/pull/664/files#diff-f2c1dbae76bd1b1341255d67aca320f1e4d1c0c4fad611e3566a02dfb8b680d1)`.

Reply:
DONE!